src/main/java/uk/gov/companieshouse/api/interceptor/CRUDAuthenticationInterceptor.java

```diff
@@ -24,11 +20,17 @@
 import uk.gov.companieshouse.logging.LoggerFactory;
 
 /**
+ * @deprecated
```

Comment:
Yes, kind of, but this has a different purpose. We're not really re-writing the CRUD interceptor, we have a different use case. This might well be valid for other types of transactional interactions. For things where the resource is being directly affected, like Follow for example, POST to create new Follow resource, DELETE will delete the user's Follow resource against that company etc. Whether Follow uses this or not, I haven't checked, it's just an example of where this type of interceptor would have fitted well. It just doesn't for our CHS filing purposes.
I'd suggest updating this comment so that it's clear what this one is for in contrast to the new Permissions interceptor you've made.

Reply:
Good idea. `MappedCRUDAuthenticationInterceptor` is intended as a like-for-like alternative to `CRUDAuthenticationInterceptor` implemented with `MappablePermissionsInterceptor` that new code may use.
I'll add DELETE to MappedCRUDAuthenticationInterceptor to fix that (it was an oversight to leave out).

That said, do you think deprecating CRUDAuthenticationInterceptor is appropriate here?

Reply:
I thought we were just writing the mappable permissions interceptor for this work and leaving the CRUD interceptor as is for the above mentioned type use cases which are aligned to that immediate type of CRUD operation. However, if the work has been done then that's fine to deprecate it. I'm struggling a little to see the benefit of the new MappedCRUD over the old CRUD implementation, some chat about that would help. I saw the two requirements as completely separate. Though like I said, if this MappedCRUD cannot be mapped further, when in use, beyond the default verbs and permissions, that's fine.

Reply:
Yes, the MappedCRUDAuthenticationInterceptor is meant as a fixed alternative to CRUDAuthenticationInterceptor, but I see your point that it's not strictly necessary.
Since we're needing this update for a story in the current sprint ending tomorrow. I'd prefer now to remove MappedCRUDAuthenticationInterceptor and not deprecate CRUDAuthenticationInterceptor.

Reply:
Yea I think that's a good idea. I'm not sure if it's solving anything and could even be considered less secure than the stand-alone version. If the Mappable Permissions class that it extends is developed further, the new MappedCRUD interceptor may inherit unwanted behaviour when it's supposed to be locked down for it's very narrowly definable purpose.

Reply:
Thanks, I've pushed an update for that if you would take a look, cheers.